### ISSUE TITLE
How to make an object heading in the direction it falls

### ISSUE THREAD

**Issue by ia97lies** [dyn4j/dyn4j#145]
I wonder how I can make a grenade (in my case this is just a rectangular physic thing) point in the direction it falls. At the moment it seems it just points in the direction I fired it in the first instance. Somehow I have to make the "head" of it a bit more heavy or so. What is the best or easiest way to achieve that?

**Comment by wnbittle** [comment]
To contrast the real world with dyn4j, there are two things that dyn4j does not simulate:

- Non-uniform mass (per body)
- Fluid resistance (air, water, etc.)

In a vacuum, a body with non-uniform mass would experience a torque applied because part of the object is experiencing a larger force.  In the presence of a fluid its even more complicated because the object can experience torques based on its shape.

In short, you need to find a way to apply a torque to the body (or just set the rotation) that simulates the real world behavior (if that's what you are going for).  For example:

```java
// run this every time step or maybe every 10 steps
Body b = //
// the initial launch direction
Vector2 launchv = //
// get the change in position from last frame
Vector2 v = b.getChangeInPosition();
// note v could be zero if the body didn't move
double a = v.getAngleBetween(launchv);
// set the rotation of the body
b.getTransform().setRotation(a);
```

This approach isn't the best.  The best approach would be to apply a torque every frame to keep the body pointed in the correct position instead of setting the rotation.  If you aren't creating a lot of these you might be able to achieve this with a WeldJoint:

```java
Body grenade = //
grenade.addFixture(Geometry.createCircle(1.0));
grenade.translate(0, 0);

// put it at the top
Body heavierPart = //
heavierPart.addFixture(Geometry.createCircle(0.25));
heavierPart.translate(0, 0.35);

Vector2 anchorPoint = new Vector2(0, 0.35);
WeldJoint j = new WeldJoint(grenade, heavierPart, anchorPoint);

world.addBody(grenade);
world.addBody(heavierPart);
world.addJoint(j);
```

(full disclosure - I have not tested this) This might give you the effect you want, but you wouldn't want to do this if everything in your scene needed this behavior.  Alternatively, you could create your own process to apply a torque to the grenade body that keeps the rotation of the body in line with it's ballistic motion.

**Comment by ia97lies** [comment]
Adding torque every frame did the trick. Just had to fiddle a bit with the amount of torque until it looks fine. I think if the grenade falls into a hole or so it will overturn. But in my case, it would be out of the screen or already detonate.

thanks a lot again for your fast reply. I wonder why I don't see your answer here?

**Comment by wnbittle** [comment]
**TLDR**: You are correct you need to apply a torque to the body to cause it to rotate about its center of mass.  Another option might be to compute the torque based on [projectile motion](https://en.wikipedia.org/wiki/Projectile_motion) equation(s).

> thanks a lot again for your fast reply. I wonder why I don't see your answer here?

Because I deleted it...  It was a terrible answer filled with incorrect information.  I was trying to complete it before I had to be somewhere and didn't think it through.  Thinking about it more while driving, I realized how bad the answer was and decided to just delete it before it did too much damage.  To amend the previous answer:

- I wrongly described gravity.  It's an acceleration applied uniformly to all bodies regardless of their mass or density distribution, therefore, it won't help here.  Recall the hammer/feather experiment on the moon.
- dyn4j _does_ support non-uniform mass via multi-fixture bodies.  But again - that doesn't help here.

The general premise was correct - to apply a torque - but the information around it was wrong - apologies.  Glad you solved it on your own despite me misleading you. :)

**Comment by ia97lies** [comment]
appreciate that! I'm glad this library is alive and well maintained. I love the improvements. And it is so super simple to integrate it into my entity component system (I use Zay-Es for that).

Keep to good work up!

**Comment by wnbittle** [comment]
Appreciate it!
